src/WebJobs.Script.WebHost/Global.asax.cs

```diff
@@ -12,13 +15,36 @@ public class WebApiApplication : System.Web.HttpApplication
     {
         protected void Application_Start()
         {
+            var settingsManager = ScriptSettingsManager.Instance;
+            var webHostSettings = WebHostSettings.CreateDefault(settingsManager);
+
+            VerifyAndEnableShadowCopy(webHostSettings);
+
             using (var metricsLogger = new WebHostMetricsLogger())
             using (metricsLogger.LatencyEvent(MetricEventNames.ApplicationStartLatency))
             {
-                GlobalConfiguration.Configure(c => WebApiConfig.Initialize(c));
+                GlobalConfiguration.Configure(c => WebApiConfig.Initialize(c, settingsManager, webHostSettings));
             }
         }
 
+        private static void VerifyAndEnableShadowCopy(WebHostSettings webHostSettings)
+        {
+            if (!FeatureFlags.IsEnabled(ScriptConstants.FeatureFlagDisableShadowCopy))
+            {
+                string currentShadowCopyDirectories = AppDomain.CurrentDomain.SetupInformation.ShadowCopyDirectories;
+                string shadowCopyPath = GetShadowCopyPath(currentShadowCopyDirectories, webHostSettings.ScriptPath);
+
+#pragma warning disable CS0618
+                AppDomain.CurrentDomain.SetShadowCopyPath(shadowCopyPath);
+#pragma warning restore CS0618
+            }
+        }
+
+        internal static string GetShadowCopyPath(string currentShadowCopyDirectories, string scriptPath)
+        {
+            return string.Join(";", new[] { currentShadowCopyDirectories, scriptPath }.Where(s => !string.IsNullOrEmpty(s)));
```

Comment:
Is this just sticking ";" + scriptPath to the end of the current string? Looks fancy :-)